pkg/usagestats/reporter.go

```diff
@@ -0,0 +1,266 @@
+				if err := rep.objectClient.DeleteObject(ctx, ClusterSeedFileName); err != nil {
+					level.Error(rep.logger).Log("msg", "failed to delete corrupted cluster seed file, deleting it", "err", err)
+				}
+				readingErr = 0
+			}
+			if continueFn == nil || continueFn(err) {
+				continue
+			}
+			return nil, err
+		}
+		return seed, nil
+	}
+	return nil, backoff.Err()
+}
+
+// readSeedFile reads the cluster seed file from the object store.
+func (rep *Reporter) readSeedFile(ctx context.Context) (*ClusterSeed, error) {
+	reader, _, err := rep.objectClient.GetObject(ctx, ClusterSeedFileName)
+	if err != nil {
+		return nil, err
+	}
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		if err := reader.Close(); err != nil {
+			level.Error(rep.logger).Log("msg", "failed to close reader", "err", err)
+		}
+	}()
+	data, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+	seed, err := JSONCodec.Decode(data)
+	if err != nil {
+		return nil, err
+	}
+	return seed.(*ClusterSeed), nil
+}
+
+// writeSeedFile writes the cluster seed to the object store.
+func (rep *Reporter) writeSeedFile(ctx context.Context, seed ClusterSeed) error {
+	data, err := JSONCodec.Encode(seed)
+	if err != nil {
+		return err
+	}
+	return rep.objectClient.PutObject(ctx, ClusterSeedFileName, bytes.NewReader(data))
+}
+
+// running inits the reporter seed and start sending report for every interval
+func (rep *Reporter) running(ctx context.Context) error {
+	rep.init(ctx)
+
+	// check every minute if we should report.
+	ticker := time.NewTicker(reportCheckInterval)
+	defer ticker.Stop()
+
+	// find  when to send the next report.
+	next := nextReport(reportInterval, rep.cluster.CreatedAt, time.Now())
+	if rep.lastReport.IsZero() {
+		// if we never reported assumed it was the last interval.
+		rep.lastReport = next.Add(-reportInterval)
+	}
+	for {
+		select {
+		case <-ticker.C:
+			now := time.Now()
+			if !next.Equal(now) && now.Sub(rep.lastReport) < reportInterval {
+				continue
+			}
+			level.Debug(rep.logger).Log("msg", "reporting cluster stats", "date", time.Now())
+			if err := rep.reportUsage(ctx, next); err != nil {
+				level.Info(rep.logger).Log("msg", "failed to report usage", "err", err)
+				continue
+			}
+			rep.lastReport = next
+			next = next.Add(reportInterval)
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+}
+
+// reportUsage reports the usage to grafana.com.
+func (rep *Reporter) reportUsage(ctx context.Context, interval time.Time) error {
+	backoff := backoff.New(ctx, backoff.Config{
+		MinBackoff: time.Second,
+		MaxBackoff: 30 * time.Second,
+		MaxRetries: 5,
+	})
+	var errs multierror.MultiError
+	for backoff.Ongoing() {
+		if err := sendReport(ctx, rep.cluster, interval); err != nil {
+			level.Info(rep.logger).Log("msg", "failed to send stats", "retries", backoff.NumRetries(), "err", err)
+			errs.Add(err)
+			backoff.Wait()
+			continue
+		}
+		level.Debug(rep.logger).Log("msg", "usage report send with success")
```

Comment:
```suggestion
			level.Info(rep.logger).Log("msg", "failed to send usage report", "retries", backoff.NumRetries(), "err", err)
			errs.Add(err)
			backoff.Wait()
			continue
		}
		level.Debug(rep.logger).Log("msg", "usage report sent with success")
```

We should keep the terminology the same so that users can debug effectively